tutorials/llm/3_filtering_history.py

```diff
@@ -0,0 +1,114 @@
+# %% [markdown]
+"""
+# LLM: 3. Filtering History
+
+If you want to take the messages that meet your particular criteria and pass them to the LLMs context you can use the `llm_response`s `filter_func` parameter.
+It must be a function that takes a single `Message` object and returns a boolean.
+"""
+
+# %pip install chatsky[llm]
+
+# %%
+from chatsky.core.message import Message
+from chatsky import (
+    TRANSITIONS,
+    RESPONSE,
+    Pipeline,
+    Transition as Tr,
+    conditions as cnd,
+    destinations as dst,
+    labels as lbl,
+)
+from chatsky.utils.testing import is_interactive_mode
+from chatsky.llm import LLM_API
+from chatsky.responses.llm import LLMResponse
+from chatsky.llm.filters import BaseFilter
+from chatsky.core.context import Context
+
+import os
+
+os.environ["OPENAI_API_KEY"] = "<TOKEN>"
+
+from langchain_openai import ChatOpenAI
+
+# %%
+model = LLM_API(
+    ChatOpenAI(model="gpt-3.5-turbo"),
+    system_prompt="You are a database assistant and must help your user to recover the demanded data from your memory.",
+)
+llm_response = LLMResponse()
+
+# %% [markdown]
+"""
+In this example we will use very simple filtering function to retrieve only the important messages.
+"""
+
+
+# %%
+class FilterImportant(BaseFilter):
```

Comment:
Add link to API ref to have a reference point for those who want to learn more about filters.